lib/master_lock/backend.rb

```diff
@@ -37,7 +37,7 @@ def synchronize(key, options = {})
       end
 
       lock = RedisLock.new(
-          redis: config.redis,
+          config: config,
```

Comment:
It looks like the only key we actually use on `config` is `redis` - should we just continue to pass in the `redis` instance instead?

Reply:
Yeah, but in `redis_lock` we call `config.key_prefix` so that's why we want to keep `config`.

Reply:
But we can go back to relying on `MasterLock.config.key_prefix` now right?

Reply:
In theory yes but feels like config is an entity that belongs to a backend. When we are doing dual writes, there are two different backends. Technically, each backend may have its own key_prefix, not that we will often see that in practice though...

Reply:
I do think we don't really need to support different key prefixes across backends, but if we want to, it might be cleaner to pass that prefix as another arg into `RedisLock` instead of making it aware of the `config` class? Either way, I don't feel too strongly here, happy to approve as is